setup.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env python
+
+# SPDX-FileCopyrightText: © 2020 Open Networking Foundation <support@opennetworking.org>
+# SPDX-License-Identifier: Apache-2.0
+
+from __future__ import absolute_import
+from setuptools import setup
+
+
+def version():
+    with open("VERSION") as f:
+        return f.read()
+
+
+def parse_requirements(filename):
+    # parse a requirements.txt file, allowing for blank lines and comments
+    requirements = []
+    for line in open(filename):
+        if line and not line.startswith("#"):
+            requirements.append(line)
+    return requirements
+
+
+setup(
+    name="onos-ric-sdk-python",
+    version=version(),
+    description="ONOS RIC SDK for Python",
+    author="Open Networking Foundation and Partners",
+    author_email="support@opennetworking.org",
+    license="Apache v2",
+    install_requires=parse_requirements("requirements.txt"),
```

Comment:
In my experience, a `requirements.txt` file is helpful because it will version pin all of the subdependencies of a project, thus creating idempotent environments for all users. That being said, if a direct dependency were to no longer need a subdependency, I don't think this would be reflected in the `requirements.txt` file unless someone created a fresh one.

Generally, I usually only see the direct dependencies listed in the `install_requires` section to accommodate for this . Another thing could be to switch to the `poetry` package which gives you the best of both worlds. Only the dependencies are listed in the `pyproject.toml` but the version pinned subdependencies are included in the `poetry.lock` file which is automatically generated.

Reply:
Currently the `requirements.txt` file is empty (save for comments), so if you want to use a different scheme, that's fine by me. 

I haven't used poetry before, partially because it would be another dependency, and also because pip supports requirements.txt already and we usually fully specify dependent package versions in that file.

Reply:
I've been using `poetry` for a bit and I've been a fan so far. Mostly for the built-in dependency resolution and also for the `poetry.lock` generation. Seems like they have support for [exporting dependencies](https://python-poetry.org/docs/cli/#export) in a `requirements.txt` format but I'm just as comfortable with `setuptools`.